File: LeetCode/136. Single Number/elixir/solution.exs
defmodule Solution do
  @spec single_number(nums :: [integer]) :: integer
  def single_number(nums) do
    nums
    |> Enum.group_by(& &1)
    |> Enum.find_value(fn
      {num, [_]} -> num
      {_, _} -> false
    end)
  end
end

File: LeetCode/136. Single Number/java/Solution.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            int val = map.getOrDefault(num, 0);
            int newVal = val + 1;
            if (newVal % 2 == 0) {
                map.remove(num);
            } else {
                map.put(num, newVal);
            }
        }
        return (int) map.keySet().toArray()[0];
    }
}